### PR TITLE
Add instructions for using docker to regen derived files

### PR DESCRIPTION
**Pull request by mhutchinson**
This is much cleaner than the previous approach of requiring protoc to be installed globally on the machine, which inevitably causes tensions with other projects that are developed on the same machine. We may want to consider building and publishing the Docker image in a repository so that people can just pull it and use it unless they are upgrading any of the tool versions (e.g. protoc).

### CODE REVIEW

File: scripts/install_deps.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+main() {
+  go install github.com/golang/mock/mockgen
+  go install google.golang.org/protobuf/proto
+  go install google.golang.org/protobuf/cmd/protoc-gen-go
+  go install google.golang.org/grpc/cmd/protoc-gen-go-grpc
+  go install github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
+  go install golang.org/x/tools/cmd/stringer
+}
+
+main "$@"
```
Comment (phbnf) [inline scripts/install_deps.sh:15]: Do you need the "$@" what would you use it for?

Comment (mhutchinson) [reply scripts/install_deps.sh]: Gone.

File: experimental/batchmap/batchmap.shims.go
[the remark this answers is not in the document]
Comment (mhutchinson) [reply experimental/batchmap/batchmap.shims.go:1]: I can put them in a separate PR, but these changes are a direct result of using the documented tooling so I think it's reasonable?

**Review by phbnf — APPROVED** [review]

File: experimental/batchmap/batchmap.shims.go
[the remark this answers is not in the document]
Comment (roger2hk) [reply experimental/batchmap/batchmap.shims.go:1]: Gotcha, this is reasonable. I didn't realise the change was generated by the tool. Normally the diff is caught by the test. Perhaps we didn't do that for experimental directory?

**Pull request merged**